dotCMS/src/integration-test/java/com/dotcms/rest/BundleResourceTest.java

```diff
@@ -223,4 +234,56 @@ public void test_downloadBundle_success() throws DotDataException {
 
         Assert.assertEquals(Status.OK.getStatusCode(),responseResource.getStatus());
     }
+
+    /**
+     * Method to Test: {@link BundleResource#generateBundle(HttpServletRequest, HttpServletResponse, AsyncResponse, GenerateBundleForm)}
+     * When: Create a bundle and generate the tar.gz file of the given bundle and verifies that the bundle contents do not contains
+     * "System Host" and references "System Folder".
+     * Should: Generate the bundle without issues, 200.
+     */
+    @Test
+    public void test_generateBundle_without_SystemHost_and_SystemFolder_references() throws Exception {
+        //Create new bundle
+        final String bundleId = insertPublishingBundle(adminUser.getUserId(),new Date());
+
+        //Create a Filter since it's needed to generate the bundle
+        createFilter();
+
+        //Generate bundle file
+        final Bundle bundle = APILocator.getBundleAPI().getBundleById(bundleId);
+        bundle.setOperation(PushPublisherConfig.Operation.PUBLISH.ordinal());
+        final File bundleFile = APILocator.getBundleAPI().generateTarGzipBundleFile(bundle);
+
+        PublisherAPIImpl.class.cast(APILocator.getPublisherAPI()).getFilterDescriptorMap().clear();
+
+        //Call download endpoint
+        final Response responseResource = bundleResource.downloadBundle(getHttpRequest(),response,bundleId);
+        Assert.assertEquals(Status.OK.getStatusCode(),responseResource.getStatus());
+
+        assertTrue(bundleFile.exists());
+
+        final Host systemHost = APILocator.getHostAPI().findSystemHost();
+        final Folder systemFolder = APILocator.getFolderAPI().findSystemFolder();
+        final String bundleDir = bundleFile.getAbsolutePath().substring(0, bundleFile.getAbsolutePath().indexOf(".tar.gz"));
+        recursiveFindAndRun(
+                Paths.get(bundleDir),
+                path -> {
+                    assertNotEquals(("/system host"), path.toFile().getName());
```

Comment:
we need the `host` extension here also, since the system_host looks like this `system host.host.xml`